go/libkb/globals.go

```diff
@@ -518,3 +530,23 @@ func (g *GlobalContext) GetUnforwardedLogger() *logger.UnforwardedLogger {
 func (g *GlobalContext) GetLog() logger.Logger {
 	return g.Log
 }
+
+func (g *GlobalContext) AddLoginHook(hook LoginHook) {
+	g.loginHooks = append(g.loginHooks, hook)
```

Comment:
Possibly.  I'll add a mutex.  I had the call to this in the service in the main goroutine originally and moved it to OnConnect.